### PR TITLE
Port deep copy from marathon-ui

### PR DESCRIPTION
**Pull request by kennyt**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @philipnrmn

File: src/js/utils/__tests__/Util-test.js [hunk 1]
```diff
@@ -243,4 +245,123 @@ describe('Util', function () {
...
+      };
+
+      var number = 83864234234;
+      originalObject.obj1.array1[number] = 3;
+
+      var copiedObject = Util.deepCopy(originalObject);
+      expect(copiedObject).not.toEqual(originalObject);
+    });
+
+    it('does clone an array with normal indices', () => {
```
Comment (MatApple) [inline src/js/utils/__tests__/Util-test.js]: Are we doing fat arrows in tests?

File: src/js/utils/__tests__/Util-test.js [hunk 2]
```diff
@@ -243,4 +245,123 @@ describe('Util', function () {
...
+        array: []
+      };
+      originalObject.array[2] = 'test';
+
+      var expectedObject = {
+        array: []
+      };
+      expectedObject.array[2] = 'test';
+
+      expect(deepEqual(Util.deepCopy(originalObject), expectedObject))
```
Comment (MatApple) [inline src/js/utils/__tests__/Util-test.js]: How come we use deepEqual for this test and not the others?

File: src/js/utils/__tests__/Util-test.js (hunk 1 above)
Comment (kennyt) [inline src/js/utils/__tests__/Util-test.js]: sorry, copied that over from marathon. just removed it

File: src/js/utils/__tests__/Util-test.js [hunk 3]
```diff
@@ -243,4 +245,123 @@ describe('Util', function () {
...
+        .toEqual(true);
+    });
+  });
```
Comment (MatApple) [inline src/js/utils/__tests__/Util-test.js]: Lets get some new lines after these

Comment (kennyt) [inline src/js/utils/__tests__/Util-test.js]: alright done

File: src/js/utils/__tests__/Util-test.js (hunk 2 above)
Comment (kennyt) [inline src/js/utils/__tests__/Util-test.js]: you're right, there's no reason to. just removed them

**Pull request merged**